net/pubsub.go

```diff
@@ -175,23 +175,15 @@ func (s *PubSub) subscribe(ctx context.Context, id thread.ID, topic *topic) {
 func (s *PubSub) handleMsg(m *pubsub.Message) (from peer.ID, rec *pb.PushRecordRequest, err error) {
 	from, err = peer.IDFromBytes(m.From)
 	if err != nil {
-		return
+		return "", nil, err
 	}
 	if from.String() == s.host.String() {
-		return
+		return "", nil, errors.New("pubsub message can not be from the host itself")
```

Comment:
From my memory, it's a good practice to return zero value when error happens, just in case if the caller doesn't check error.